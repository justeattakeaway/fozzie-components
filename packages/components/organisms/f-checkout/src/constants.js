export const CHECKOUT_METHOD_COLLECTION = 'collection';
export const CHECKOUT_METHOD_DELIVERY = 'delivery';

export const VALID_CHECKOUT_METHODS = [CHECKOUT_METHOD_COLLECTION, CHECKOUT_METHOD_DELIVERY];

export const TENANT_MAP = {
    'en-GB': 'uk',
    'en-AU': 'au',
    'en-NZ': 'nz',
    'da-DK': 'dk',
    'es-ES': 'es',
    'en-IE': 'ie',
    'it-IT': 'it',
    'nb-NO': 'no'
};

export const VALIDATIONS = {
    address: 'addressValidations',
    guest: 'guestValidations'
};
