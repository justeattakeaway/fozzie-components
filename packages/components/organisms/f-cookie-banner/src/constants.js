export const CONSENT_COOKIE_NAME = 'je-cookieConsent';
export const LEGACY_CONSENT_COOKIE_NAME = 'je-banner_cookie';
