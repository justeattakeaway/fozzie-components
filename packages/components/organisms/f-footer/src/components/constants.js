export const confianzaUrl = 'https://www.confianzaonline.es/empresas/justeatspain.htm';
export const confianzaScreenReaderText = 'Accede a Confianza Online y comprueba nuestro sello de calidad en Internet';
