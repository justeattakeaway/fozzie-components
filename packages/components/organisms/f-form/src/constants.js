export const FORM_EVENTS = {
    fieldUpdated: 'updated',
    submitting: 'form-submitting'
};

export const PROP_VALIDATION_MESSAGES = {
    requiredFormFields: '`formFields` Not found in `formData` prop, please add an array of `formFields`',
    invalidFormFields: '`formFields` are invalid, please add an array of formFields',
    requiredNameProperty: 'Each field in `formFields` should contain "name" property',
    requiredTranslationsProperty: 'Each field in `formFields` should contain "translations" property',
    requiredLabelProperty: 'Each field in `formFields` should contain "translations.label" property'
};

export const DEFAULT_BUTTON_TEXT = 'Submit';
