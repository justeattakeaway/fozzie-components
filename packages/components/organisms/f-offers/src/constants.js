export const POSTCODE_EMPTY = 'POSTCODE_EMPTY';
export const POSTCODE_INVALID = 'POSTCODE_INVALID';
