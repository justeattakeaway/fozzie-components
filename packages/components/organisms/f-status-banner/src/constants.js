/* eslint-disable import/prefer-default-export */
export const BASE_HERO_URL = 'https://d30v2pzvrfyzpo.cloudfront.net/status-banner/';
