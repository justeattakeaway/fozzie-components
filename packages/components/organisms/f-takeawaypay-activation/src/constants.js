export const ACTIVATION_STATE_NONE = 'none';
export const ACTIVATION_STATE_AVAILABLE_LOGGED_IN = 'available_logged_in';
export const ACTIVATION_STATE_AVAILABLE_NOT_LOGGED_IN = 'available_not_logged_in';
export const ACTIVATION_STATE_SUCCEEDED = 'succeeded';
export const ACTIVATION_STATE_FAILED = 'failed';
