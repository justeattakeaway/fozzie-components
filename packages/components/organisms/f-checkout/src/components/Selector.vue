<template>
    <form-dropdown
        id="time-selection"
        name="order-time"
        :class="$style['c-checkout-selector']"
        input-type="dropdown"
        :label-text="orderMethod"
        :dropdown-options="fulfilmentTimes"
        @input="selectionChanged" />
</template>

<script>
import { mapActions, mapState } from 'vuex';
import FormDropdown from '@justeat/f-form-field';
import '@justeat/f-form-field/dist/f-form-field.css';
import { CHECKOUT_METHOD_DELIVERY } from '../constants';

export default {
    components: { FormDropdown },

    data () {
        return {
            selectedAvailableFulfilmentTime: ''
        };
    },

    computed: {
        ...mapState('checkout', [
            'availableFulfilment',
            'serviceType'
        ]),

        orderMethod () {
            return this.serviceType === CHECKOUT_METHOD_DELIVERY
                ? this.$t('labels.deliveryOrderMethod')
                : this.$t('labels.collectionOrderMethod');
        },

        /*
        * Returns an array of formatted dates to display as options in a dropdown
        */
        fulfilmentTimes () {
            const times = [];
            this.availableFulfilment.times.forEach(time => {
                const formattedTime = this.$d(new Date(time.from), 'short');

                times.push({
                    text: formattedTime,
                    value: time.from
                });
            });

            if (this.availableFulfilment.isAsapAvailable && times.length) {
                times[0].text = this.$t('asapFulfilmentOption');
            }

            return times;
        }
    },

    watch: {
        fulfilmentTimes (newFulfilmentTimes) {
            this.selectionChanged(newFulfilmentTimes[0] && newFulfilmentTimes[0].value);
        }
    },

    methods: {
        ...mapActions('checkout', [
            'updateFulfilmentTime'
        ]),

        ...mapActions('analytics', [
            'updateChangedField'
        ]),

        /**
        * Update the selected available fulfilment time.
        *
        * @param {string} selectedFulfilmentTime The time emited when dropdown value is changed.
        **/
        selectionChanged (selectedFulfilmentTime) {
            this.selectedAvailableFulfilmentTime = selectedFulfilmentTime;
            this.updateChangedField('orderTime');

            // TODO - Update to use different from/to times when the API supports it
            this.updateFulfilmentTime({
                from: selectedFulfilmentTime,
                to: selectedFulfilmentTime
            });
        }
    }
};
</script>

<style lang="scss" module>
.c-checkout-selector {
    margin-top: spacing(x2);
}
</style>
