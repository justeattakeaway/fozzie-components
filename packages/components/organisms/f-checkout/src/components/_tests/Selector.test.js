import Vuex from 'vuex';
import { VueI18n } from '@justeat/f-globalisation';
import { shallowMount, createLocalVue, mount } from '@vue/test-utils';
import { CHECKOUT_METHOD_DELIVERY, CHECKOUT_METHOD_COLLECTION } from '../../constants';
import Selector from '../Selector.vue';
import {
    fulfilmentTimes, defaultCheckoutState, i18n, createStore
} from './helpers/setup';

const localVue = createLocalVue();

localVue.use(VueI18n);
localVue.use(Vuex);

describe('Selector', () => {
    allure.feature('Checkout-Selector');
    const propsData = {};

    it('should be defined', () => {
        // Arrange & Act
        const wrapper = shallowMount(Selector, {
            store: createStore(),
            i18n,
            localVue,
            propsData
        });

        // Assert
        expect(wrapper.exists()).toBe(true);
    });

    describe('computed ::', () => {
        describe('orderMethod', () => {
            it('should show the delivery label when the `serviceType` is `delivery`', () => {
                // Arrange & Act
                const wrapper = mount(Selector, {
                    store: createStore({ ...defaultCheckoutState, serviceType: CHECKOUT_METHOD_DELIVERY }),
                    i18n,
                    localVue,
                    propsData
                });

                const selector = wrapper.find('[data-test-id="formfield-order-time-dropdown-select"]');

                // Assert
                expect(selector.html()).toMatchSnapshot();
            });

            it('should show the collection label when the `serviceType` is `collection`', () => {
                // Arrange & Act
                const wrapper = mount(Selector, {
                    store: createStore({ ...defaultCheckoutState, serviceType: CHECKOUT_METHOD_COLLECTION }),
                    i18n,
                    localVue,
                    propsData
                });

                const selector = wrapper.find('[data-test-id="formfield-order-time-dropdown-select"]');

                // Assert
                expect(selector.html()).toMatchSnapshot();
            });
        });

        describe('fulfilmentTimes', () => {
            it('should create an array of formatted fulfilment times', () => {
                // Arrange & Act
                const wrapper = shallowMount(Selector, {
                    store: createStore({ ...defaultCheckoutState }),
                    i18n,
                    localVue,
                    propsData
                });

                const expectedTimes = [
                    { text: 'As soon as possible', value: '2020-01-01T01:00:00.000Z' },
                    { text: 'Wednesday 01:15', value: '2020-01-01T01:15:00.000Z' }
                ];

                // Assert
                expect(wrapper.vm.fulfilmentTimes).toEqual(expectedTimes);
            });

            it('should create an array of formatted fulfilment times without `as soon as possible` if `isAsapAvailable` is `false`', () => {
                // Arrange & Act
                const wrapper = shallowMount(Selector, {
                    store: createStore({
                        ...defaultCheckoutState,
                        availableFulfilment: {
                            times: fulfilmentTimes,
                            isAsapAvailable: false
                        }
                    }),
                    i18n,
                    localVue,
                    propsData
                });

                const expectedTimes = [
                    { text: 'Wednesday 01:00', value: '2020-01-01T01:00:00.000Z' },
                    { text: 'Wednesday 01:15', value: '2020-01-01T01:15:00.000Z' }
                ];

                // Assert
                expect(wrapper.vm.fulfilmentTimes).toEqual(expectedTimes);
            });

            it('should create an empty array when there are no times available and `isAsapAvailable` is `false`', () => {
                // Arrange & Act
                const wrapper = shallowMount(Selector, {
                    store: createStore({
                        ...defaultCheckoutState,
                        availableFulfilment: {
                            times: [],
                            isAsapAvailable: false
                        }
                    }),
                    i18n,
                    localVue,
                    propsData
                });

                const expectedTimes = [];

                // Assert
                expect(wrapper.vm.fulfilmentTimes).toEqual(expectedTimes);
            });
        });
    });

    describe('methods ::', () => {
        describe('selectionChanged', () => {
            const field = 'orderTime';
            const selectedTime = '2020-01-01T01:00:00.000Z';

            let wrapper;
            let updateChangedFieldSpy;

            beforeEach(() => {
                updateChangedFieldSpy = jest.spyOn(Selector.methods, 'updateChangedField');

                wrapper = shallowMount(Selector, {
                    store: createStore(),
                    i18n,
                    localVue,
                    propsData
                });
            });

            it('should call `updateChangedField` with `orderTime`', () => {
                // Act
                wrapper.vm.selectionChanged(selectedTime);

                // Assert
                expect(updateChangedFieldSpy).toHaveBeenCalledWith(field);
            });

            it('should update `selectedAvailableFulfilmentTime` with the value passed', () => {
                // Act
                wrapper.vm.selectionChanged(selectedTime);

                // Assert
                expect(wrapper.vm.selectedAvailableFulfilmentTime).toBe(selectedTime);
            });
        });
    });

    describe('watch ::', () => {
        describe('fulfilmentTimes ::', () => {
            afterEach(() => {
                jest.clearAllMocks();
            });
            it('should call `selectionChanged` with the first fulfilment time when there are fulfilment times', () => {
                // Arrange
                const expected = '2020-01-01T01:00:00.000Z';
                const selectionChangedSpy = jest.spyOn(Selector.methods, 'selectionChanged');

                const wrapper = shallowMount(Selector, {
                    store: createStore(),
                    i18n,
                    localVue,
                    propsData
                });

                // Act
                wrapper.vm.$options.watch.fulfilmentTimes.call(wrapper.vm, [
                    {
                        text: 'Wednesday 01:00',
                        value: '2020-01-01T01:00:00.000Z'
                    }
                ]);

                // Assert
                expect(selectionChangedSpy).toHaveBeenCalledWith(expected);
            });

            it('should call `selectionChanged` with `undefined` when there are no fulfilment times', () => {
                // Arrange
                const selectionChangedSpy = jest.spyOn(Selector.methods, 'selectionChanged');

                const wrapper = shallowMount(Selector, {
                    store: createStore({
                        ...defaultCheckoutState,
                        availableFulfilment: {
                            isAsapAvailable: false,
                            times: []
                        },
                        serviceType: CHECKOUT_METHOD_DELIVERY
                    }),
                    i18n,
                    localVue,
                    propsData
                });

                // Act
                wrapper.vm.$options.watch.fulfilmentTimes.call(wrapper.vm, []);

                // Assert
                expect(selectionChangedSpy).toHaveBeenCalledWith(undefined);
            });
        });
    });
});
