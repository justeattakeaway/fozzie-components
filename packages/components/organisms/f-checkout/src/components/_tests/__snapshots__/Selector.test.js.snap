// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Selector computed :: orderMethod should show the collection label when the \`serviceType\` is \`collection\` 1`] = `
<select id="formField-order-time" data-test-id="formfield-order-time-dropdown-select" name="order-time" class="FormDropdown_c-formDropdown-select_ROvjN">
  <option data-test-id="formfield-order-time-dropdown-option-0" value="2020-01-01T01:00:00.000Z">As soon as possible</option>
  <option data-test-id="formfield-order-time-dropdown-option-1" value="2020-01-01T01:15:00.000Z">Wednesday 01:15</option>
</select>
`;

exports[`Selector computed :: orderMethod should show the delivery label when the \`serviceType\` is \`delivery\` 1`] = `
<select id="formField-order-time" data-test-id="formfield-order-time-dropdown-select" name="order-time" class="FormDropdown_c-formDropdown-select_ROvjN">
  <option data-test-id="formfield-order-time-dropdown-option-0" value="2020-01-01T01:00:00.000Z">As soon as possible</option>
  <option data-test-id="formfield-order-time-dropdown-option-1" value="2020-01-01T01:15:00.000Z">Wednesday 01:15</option>
</select>
`;
